soh/soh/Enhancements/crowd-control/CrowdControl.cpp

```diff
@@ -265,31 +265,34 @@ CrowdControl::Effect* CrowdControl::ParseMessage(char payload[512]) {
     // Assign GameInteractionEffect + values to CC effect.
     // Categories are mostly used for checking for conflicting timed effects.
     if (effectName == EFFECT_ADD_HEART_CONTAINER) {
-        effect->giEffect = new GameInteractionEffect::AddHeartContainers();
+        effect->giEffect = new GameInteractionEffect::ModifyHeartContainers();
         effect->giEffect->parameter = 1;
     } else if (effectName == EFFECT_REMOVE_HEART_CONTAINER) {
-        effect->giEffect = new GameInteractionEffect::RemoveHeartContainers();
-        effect->giEffect->parameter = 1;
+        effect->giEffect = new GameInteractionEffect::ModifyHeartContainers();
+        effect->giEffect->parameter = -1;
     } else if (effectName == EFFECT_FILL_MAGIC) {
         effect->giEffect = new GameInteractionEffect::FillMagic();
     } else if (effectName == EFFECT_EMPTY_MAGIC) {
         effect->giEffect = new GameInteractionEffect::EmptyMagic();
     } else if (effectName == EFFECT_ADD_RUPEES) {
-        effect->giEffect = new GameInteractionEffect::GiveRupees();
+        effect->giEffect = new GameInteractionEffect::ModifyRupees();
     } else if (effectName == EFFECT_REMOVE_RUPEES) {
-        effect->giEffect = new GameInteractionEffect::TakeRupees();
+        effect->giEffect = new GameInteractionEffect::ModifyRupees();
+        effect->paramMultiplier = -1;
```

Comment:
Couldn't we assign effect->value[0] and invert it here instead of introducing a paramMultiplier?

Reply:
The catch-all wouldn't change if we already set the parameter inside of here. I'm not super opposed to doing it this way, it just felt a bit overkill to set up a new property and adjust the catch-all to accomodate for negative numbers. Your call honestly.